src/kernel/capinv.c

```diff
@@ -1147,19 +1164,29 @@ static int __attribute__((noinline)) composite_syscall_slowpath(struct pt_regs *
 		case CAPTBL_OP_THDACTIVATE: {
 			thdclosure_index_t init_data  = __userregs_get1(regs) >> 16;
 			capid_t thd_cap               = __userregs_get1(regs) & 0xFFFF;
-			capid_t pgtbl_cap             = __userregs_get2(regs);
+			capid_t pgtbl_cap             = __userregs_get2(regs) >> 16;
```

Comment:
Looks pretty concerning to me. especially since we need to pass in dcbinfo as well. It's going to be very ugly. ;(